test/blocklib/algorithm/fft/fft.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef GRAPH_PROTOTYPE_ALGORITHM_FFT_HPP
+#define GRAPH_PROTOTYPE_ALGORITHM_FFT_HPP
+
+#include "../../core/fft/window.hpp"
+#include "fft_types.hpp"
+#include "node.hpp"
+#include <ranges>
+
+namespace gr::algorithm {
+
+template<typename T>
+    requires(ComplexType<T> || std::floating_point<T>)
+struct FFT {
+    using PrecisionType = typename FFTInternal<T>::PrecisionType;
```

Comment:
I put most of the code related to the FFT types to the separate file because it is used for both `FFT` and `FFTw` algorithms to get the proper input, output, precision etc. types. Basically now the types are defined automatically. But even if we give possibility to define both input and output types to the user, we need to add error/compatibility checker which is basically how it is implemented now.